src/aggregator/client/conn.go

```diff
@@ -199,9 +201,13 @@ func (c *connection) checkReconnectWithLock() error {
 	// If we haven't accumulated enough failures to warrant another reconnect
 	// and we haven't past the maximum duration since the last time we attempted
 	// to connect then we simply return false without reconnecting.
-	tooManyFailures := c.numFailures >= c.threshold
+	// If we exhausted maximum allowed failures then we will retry only based on
+	// maximum duration since the last attempt.
+	enoughFailuresToRetry := c.numFailures >= c.threshold
+	exhaustedMaxFailures := c.numFailures >= c.maxThreshold
 	sufficientTimePassed := c.nowFn().UnixNano()-c.lastConnectAttemptNanos >= c.maxDuration.Nanoseconds()
-	if !tooManyFailures && !sufficientTimePassed {
+	if (!enoughFailuresToRetry && !sufficientTimePassed) ||
+		(exhaustedMaxFailures && !sufficientTimePassed) {
```

Comment:
Both of these cases use `!sufficientTimePassed`, so you can write this as:

```go
if !sufficientTimePassed && (exhaustedMaxFailures || !enoughFailuresToRetry) {
  // ...
}
```

Reply:
yep, thanks